haystack/document_stores/search_engine.py

```diff
@@ -415,31 +415,36 @@ def write_documents(
         )
         documents_to_index = []
         for doc in document_objects:
-            _doc = {
+            index_message: Dict[str, Any] = {
                 "_op_type": "index" if duplicate_documents == "overwrite" else "create",
                 "_index": index,
-                **doc.to_dict(field_map=self._create_document_field_map()),
-            }  # type: Dict[str, Any]
+                "_id": str(doc.id),
+            }
+
+            _source: Dict[str, Any] = doc.to_dict(field_map=self._create_document_field_map())
 
             # cast embedding type as ES cannot deal with np.array
-            if _doc[self.embedding_field] is not None:
-                if type(_doc[self.embedding_field]) == np.ndarray:
-                    _doc[self.embedding_field] = _doc[self.embedding_field].tolist()
+            if _source[self.embedding_field] is not None:
+                if type(_source[self.embedding_field]) == np.ndarray:
+                    _source[self.embedding_field] = _source[self.embedding_field].tolist()
 
-            # rename id for elastic
-            _doc["_id"] = str(_doc.pop("id"))
+            # we already have the id in the index message
+            _source.pop("id")
 
             # don't index query score and empty fields
-            _ = _doc.pop("score", None)
-            _doc = {k: v for k, v in _doc.items() if v is not None}
+            _source.pop("score", None)
+            _source = {k: v for k, v in _source.items() if v is not None}
 
             # In order to have a flat structure in elastic + similar behaviour to the other DocumentStores,
             # we "unnest" all value within "meta"
-            if "meta" in _doc.keys():
-                for k, v in _doc["meta"].items():
-                    _doc[k] = v
-                _doc.pop("meta")
-            documents_to_index.append(_doc)
+            if "meta" in _source.keys():
```

Comment:
What about:

```
_source.update(_source.pop("meta", {}))
``` 

Would that work?

Reply:
Have to take care of `_source["meta] == None`. Besides that it would work.